### ISSUE TITLE
[feature-request] promisifyAll allow full control over promisified method name

### ISSUE THREAD

**Issue by mbroadst** [petkaantonov/bluebird#1261]
1) What version of bluebird is the issue happening on?
v3.4.6

2) What platform and version? (For example Node.js 0.12 or Google Chrome 32)
node.js 6.5.x

3) Did this issue happen with earlier version of bluebird?
yes

It would be useful to have more control over the generated name when using `promisifyAll`. Currently we are allowed to specify a suffix, however in our project we wanted to use `promisifyAll` to mutate methods exposed by a native module in the form `_method` such that the promisified version would be `method`. As far as I'm looking at the code, this isn't possible but should be relatively easy to add - if you're amenable I could submit a PR.

**Comment by benjamingr** [comment]
It would make sense for this to replace `suffix` if we add it and have `suffix` use it instead, no?

**Comment by mbroadst** [comment]
@benjamingr agreed, I didn't want to break comparability though. Whatever you all think is best.

**Comment by mbroadst** [comment]
@benjamingr Oh I see, sorry was reading on my phone before. You just mean that `suffix` itself should use the `nameTranslator`. I can implement that in my PR as well if you'd like?

**Comment by mbroadst** [comment]
@benjamingr so I've done a little bit of exploratory work to implement that change, and have run into a bit of a snag. Part of the promisification involves checking the validity of the newly created name: https://github.com/petkaantonov/bluebird/blob/master/src/promisify.js#L56. This works because suffix is presently just that: a suffix, and therefore the operation is deterministic. I'm not quite sure how we could provide the same support when providing a function that allows all sorts of changes to the original name?
